pkg/cdi/cache.go

```diff
@@ -107,9 +109,7 @@ func (c *Cache) configure(options ...Option) error {
 		c.watch.setup(c.specDirs, c.dirErrors)
 		c.watch.start(&c.Mutex, c.refresh, c.dirErrors)
 	}
-	c.refresh()
-
-	return nil
+	return c.refresh()
```

Comment:
TBH, off the top of my head I'm not sure if this could cause problems.

Looking at the code, maybe it would be better/simpler/cleaner to change the API so that [Rr]efresh() never returns an error. Since errors encountered during the last refresh can anyway be queried using GetErrors(), if a caller is interested in errors it can query them itself.